File: HRMS/HRMS.Web/Modules/Communication/Notice/NoticeColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Columns;

[ColumnsScript("Communication.Notice")]
[BasedOnRow(typeof(NoticeRow), CheckNames = true)]
public class NoticeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int NoticeId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public NoticePriority Priority { get; set; }
    public DateTime PublishDate { get; set; }
    public DateTime ExpiryDate { get; set; }
    public bool IsActive { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/NoticeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { NoticePriority } from "./NoticePriority";
import { NoticeRow } from "./NoticeRow";

export interface NoticeColumns {
    NoticeId: Column<NoticeRow>;
    Title: Column<NoticeRow>;
    Priority: Column<NoticeRow>;
    PublishDate: Column<NoticeRow>;
    ExpiryDate: Column<NoticeRow>;
    IsActive: Column<NoticeRow>;
}

export class NoticeColumns extends ColumnsBase<NoticeRow> {
    static readonly columnsKey = 'Communication.Notice';
    static readonly Fields = fieldsProxy<NoticeColumns>();
}

[NoticePriority]; // referenced types